CONTRIBUTING.md

```diff
@@ -209,6 +209,52 @@ for (const blockResult of blocksResult) {
 }
 ```
 
+## Managing and Opening Feature/Large PRs
+
+To maintain code quality, improve collaboration, and ensure clarity in large or complex changes, we follow these guidelines when opening pull requests (PRs). Depending on the nature of the change, PRs fall into three categories:
+
+### 1. Single, Complete PR
+
+If the PR contains a self-contained and complete feature or bug fix that does not require major refactoring or cross-team discussions, then:
+
+- Clearly explain the rationale and motivation behind the change in the PR description.
+- Provide relevant context, including:
+- Problem the PR is solving.
+- Why the approach was chosen.
+- Any alternatives considered.
+- If the PR modifies critical code paths, add references to relevant issues, benchmarks, or related discussions.
+- Ensure the PR adheres to our standard PR etiquette and commit message guidelines.
+
+### 2. PR with Major Refactoring
+
+If the PR involves significant code refactoring, structural changes, or fundamental modifications where team input is needed:
+
+- Start a GitHub Discussion or Discord thread before writing code.
```

Comment:
I would suggest that we do Discord for brainstorm back and forth and use Github Discussions to persist the outcome of what was discussed on the Discord thread.

Reply:
I thought it's fine to have both options (as per https://github.com/ChainSafe/lodestar/pull/7602#discussion_r2010328512) but personally lean more towards discord for those kind of early (and potentially chaotic) discussions